summingbird-core/src/main/scala/com/twitter/summingbird/memory/Memory.scala

```diff
@@ -82,11 +116,11 @@ class Memory(implicit jobID: JobId = JobId("default.memory.jobId")) extends Plat
           case AlsoProducer(l, r) =>
             //Plan the first one, but ignore it
             val (left, leftM) = toStream(l, jamfs)
-            val (right, rightM) = toStream(r, leftM)
             // We need to force all of left to make sure any
             // side effects in write happen
-            lazy val lforcedEmpty = left.filter(_ => false)
-            (right.append(lforcedEmpty), rightM)
+            val lforcedEmpty = left.filter(_ => false)
+            val (right, rightM) = toStream(r, leftM)
```

Comment:
just to link to the prior discussion:
https://github.com/twitter/summingbird/pull/647/files#r47308385

#647 may indeed be fixed more than one way, but I still don't like that planning forces the left hand side.

Could you get the tests to pass if you did something like:
```scala
def merge[A, B](a: Stream[A], b: Stream[B]): Stream[Either[A, B]] = {
  val itA = a.iterator
  val itB = b.iterator
  var left = itA.hasNext
    // try to alternate as much as possible
  def next: Option[Either[A, B]] =
    if (!(itA.hasNext || itB.hasNext)) None
    else if (left) {
      left = false
      if (itA.hasNext) Some(Left(itA.next)) else next
    } else {
      left = true
      if (itB.hasNext) Some(Right(itB.next)) else next
    }
  Stream.continually(next).takeWhile(_.isDefined).map(_.get)
}
```
Then we can return `merge(left, right).collect { case Right(r) => r }`. as the result of the `Also`. In this way, we are closer to modeling a real streaming system.

If the contract we are presenting here is that the left will always be consumed at planning, I just don't like that. It leads to a side-effect from planning which was never our intent. I think the test needs to not assume that.

I think we need to make a clear case as to why the behavior is a certain way to which is not "some broken test at Twitter will fail if we change this". If that is the only reason, you should have an internal fork of the platform.